packetbeat/beater/packetbeat.go

```diff
@@ -109,50 +109,50 @@ func (pb *Packetbeat) Config(b *beat.Beat) error {
 		return err
 	}
 
+	cfg := &pb.PbConfig.Packetbeat
+
 	// CLI flags over-riding config
 	if *pb.CmdLineArgs.TopSpeed {
-		pb.PbConfig.Interfaces.TopSpeed = true
+		cfg.Interfaces.TopSpeed = true
 	}
 
 	if len(*pb.CmdLineArgs.File) > 0 {
-		pb.PbConfig.Interfaces.File = *pb.CmdLineArgs.File
+		cfg.Interfaces.File = *pb.CmdLineArgs.File
 	}
 
-	pb.PbConfig.Interfaces.Loop = *pb.CmdLineArgs.Loop
-	pb.PbConfig.Interfaces.OneAtATime = *pb.CmdLineArgs.OneAtAtime
+	cfg.Interfaces.Loop = *pb.CmdLineArgs.Loop
+	cfg.Interfaces.OneAtATime = *pb.CmdLineArgs.OneAtAtime
 
 	if len(*pb.CmdLineArgs.Dumpfile) > 0 {
-		pb.PbConfig.Interfaces.Dumpfile = *pb.CmdLineArgs.Dumpfile
+		cfg.Interfaces.Dumpfile = *pb.CmdLineArgs.Dumpfile
 	}
 
-	// assign global singleton as it is used in protocols
-	// TODO: Refactor
-	config.ConfigSingleton = pb.PbConfig
-
 	return nil
 }
 
 // Setup packetbeat
 func (pb *Packetbeat) Setup(b *beat.Beat) error {
 
-	if err := procs.ProcWatcher.Init(pb.PbConfig.Procs); err != nil {
+	cfg := &pb.PbConfig.Packetbeat
+
+	if err := procs.ProcWatcher.Init(cfg.Procs); err != nil {
 		logp.Critical(err.Error())
 		return err
 	}
 
 	queueSize := defaultQueueSize
-	if pb.PbConfig.Shipper.QueueSize != nil {
-		queueSize = *pb.PbConfig.Shipper.QueueSize
+	if cfg.Shipper.QueueSize != nil {
+		queueSize = *cfg.Shipper.QueueSize
```

Comment:
This means that these two options under `shipper` now are under `packetbeat.shipper`. Since these settings might be read by both Packetbeat and libbeat, but now they are defined in two parts. We could either leave Shipper top level in `config.Config` or clean this up to not require Packetbeat reading options from the libbeat sections.